src/components/ConnectionsListProfiles.tsx

```diff
@@ -15,18 +20,32 @@ enum ListStatus {
 }
 
 interface ConnectionsListProfilesProps {
+  userId: DSNPUserId;
   listStatus: ListStatus;
-  following: Record<DSNPUserId, boolean>;
-  followers: Record<DSNPUserId, boolean>;
+  following: Record<DSNPUserId, RelationshipStatus>;
+  followers: Record<DSNPUserId, RelationshipStatus>;
 }
 
 const ConnectionsListProfiles = ({
+  userId,
   listStatus,
   following,
   followers,
 }: ConnectionsListProfilesProps): JSX.Element => {
-  const userRelationship = (userProfile: Profile) =>
-    following[userProfile.fromId] ? "Unfollow" : "Follow";
+  const dispatch = useAppDispatch();
+
+  const isFollowing = (userProfile: Profile): boolean =>
+    userProfile.fromId in following;
+
+  const isFollowingUpdating = (userProfile: Profile): boolean =>
+    following[userProfile.fromId] === RelationshipStatus.UPDATING;
+
+  const userRelationship = (profile: Profile): string =>
```

Comment:
Same here as below. `userRelationship` is a bit unclear. `connectionRelationship` perhaps? Or just `relationshipStr` even?